src/server/handlers/github.js

```diff
@@ -100,9 +101,8 @@ export const getSnapName = (repositoryUrl, token) => {
         .then((snapcraftYaml) => {
           const snapName = snapcraftYaml.name;
           if (snapName) {
-            return getMemcached().set(cacheId, snapName, 3600, () => {
-              return resolve(snapName);
-            });
+            getMemcached().set(cacheId, snapName, 3600);
+            return resolve(snapName);
```

Comment:
Memcached is stubbed when not configured (https://github.com/canonical-ols/build.snapcraft.io/blob/master/src/server/helpers/memcached.js#L15) so it should always call the callback.

Maybe something else was causing an issue?